src/compas/datastructures/mesh/mesh_to_points.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from numpy import array
+from numpy.random import choice
+from numpy.random import rand
+from numpy import sqrt
+
+
+
+
+__all__ = [
+    'mesh_to_points',
+]
+
+def mesh_to_points(mesh, num_points: int = 10000):
+    """Convert a mesh to points
+
+    Parameters
+    ----------
+    mesh : compas.datastructures.Mesh
+        The mesh data structure.
+    num_points : (int) 
+        How many points sampled
+    return_normals : (bool)  (TODO)
+        Return normals for the points, if True
+
+    Returns
+    -------
+    list
+        A numpy array with dim = [num_points, 3].
+
+    Examples
+    --------
+
+
+    """
+    if mesh.is_empty():
+    raise ValueError("Mesh is empty.")
+    if mesh.is_quadmesh():
+        mesh.quads_to_triangles()
+    if not mesh.is_valid():
+        raise ValueError("Mesh is invalid.")
+        
+    # (1)  acquire the list of area
+    first_idx = list(mesh.face.keys())[0]
+    add_idx = 0
+    area_list=[]
+    if first_idx != 0:
+        add_idx = first_idx
+    for i in range(len(mesh.face)):
+        area_list.append(mesh.face_area(i+add_idx))
+    area_list = array(area_list)
+    area_list_norm = area_list / area_list.sum()
+
+    # (2) sample num_points on a mesh face regarding its weight of area
+    mesh_face_idx = list(mesh.face.keys())
+    sample_face_idxs = choice(mesh_face_idx, num_points, p=area_list_norm) - add_idx
+
+    # (3) create a ndarray of vertices regarding the faces
+    vertices = mesh.to_vertices_and_faces()[0]
+    vertices = array(vertices)
+    faces = mesh.to_vertices_and_faces()[1]
+    faces = array(faces)
+    faces_vertices = vertices[faces]
```

Comment:
this can be done in one step.
for large meshes it will make a difference...

```python
# 3
vertices, faces = mesh.to_vertices_and_faces()
vertices = array(vertices, dtype=numpy.float64)
faces = array(faces)
faces_vertices = vertices[faces]
```